src/Lifecycle.js

```diff
@@ -196,26 +200,26 @@ function _registerAsGuest(hsUrl, isUrl, defaultDeviceDisplayName) {
 //      SessionStore to avoid bugs where the view becomes out-of-sync with
 //      localStorage (e.g. teamToken, isGuest etc.)
 function _restoreFromLocalStorage() {
-    if (!localStorage) {
-        return Promise.resolve(false);
-    }
-    const hsUrl = localStorage.getItem("mx_hs_url");
-    const isUrl = localStorage.getItem("mx_is_url") || 'https://matrix.org';
-    const accessToken = localStorage.getItem("mx_access_token");
-    const userId = localStorage.getItem("mx_user_id");
-    const deviceId = localStorage.getItem("mx_device_id");
-
-    let isGuest;
-    if (localStorage.getItem("mx_is_guest") !== null) {
-        isGuest = localStorage.getItem("mx_is_guest") === "true";
-    } else {
-        // legacy key name
-        isGuest = localStorage.getItem("matrix-is-guest") === "true";
-    }
+    return Promise.resolve().then(() => {
```

Comment:
I'm finding it quite strange that the whole thing is wrapped in a resolved promise. Should this be an async function?